tendermint/src/rpc/endpoint/commit.rs

```diff
@@ -46,3 +46,26 @@ pub struct SignedHeader {
     /// Commit containing signatures for the header
     pub commit: block::Commit,
 }
+
+impl lite::Commit for SignedHeader {
+    type Vote = SignedVote;
+    fn header_hash(&self) -> Hash {
+        self.commit.block_id.hash
+    }
+    fn into_vec(&self) -> Vec<Option<Self::Vote>> {
+        let chain_id = self.header.chain_id.to_string();
+        let mut votes = self.commit.precommits.clone().into_vec();
+        votes
+            .drain(..)
```

Comment:
@yihuang Why use `drain` here rather than `iter` ?

Reply:
Yeah, my fault, not familiar enough with idiomatic rust then. But I believe this code already modified by @Liamsi in the final merged version ;D